sa/sa.go

```diff
@@ -1272,3 +1273,44 @@ func (ssa *SQLStorageAuthority) DeactivateAuthorization(ctx context.Context, id
 	}
 	return tx.Commit()
 }
+
+// NewOrder adds a new v2 style order to the database
+func (ssa *SQLStorageAuthority) NewOrder(ctx context.Context, req *corepb.Order) (*corepb.Order, error) {
+	order := &orderModel{
+		RegistrationID: *req.RegistrationID,
+		Expires:        time.Unix(0, *req.Expires),
+		CSR:            req.Csr,
+		Status:         core.AcmeStatus(*req.Status),
+	}
+
+	tx, err := ssa.dbMap.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.Insert(order)
+	if err != nil {
+		err = Rollback(tx, err)
+		return nil, err
+	}
+
+	for _, authz := range req.Authorizations {
```

Comment:
Note: This will create a large number of DB round-trips for certificates with large numbers of names. If we provide a list of objects to `tx.Insert`, will that batch them into a single statement, or will we still have the round-trip issue?